lib/resty/upload.lua

```diff
@@ -28,7 +27,7 @@ local state_handlers
 
 
 local function get_boundary()
-    local header = get_headers()["content-type"]
+    local header = ngx.var.http_content_type
```

Comment:
I'm fine with switching over to reading the `$http_content_type` nginx variable.

But maybe we could cache the `ngx.var` table in an outer scope so that we don't have to look up the `var` field in the `ngx` table on this hot code path?